### PR TITLE
chore: refactor curl import API to support context type

### PR DESCRIPTION
**Pull request by nsarupr**
## Description

Support curl imports for different contexts. 

### Server changes

Previous API: 
`/api/v1/import?type=CURL&pageId={pageId}&name=Api2&workspaceId={workspaceId}`

New API: 
With context type, it will create for the specific context.
`/api/v1/import?type=CURL&contextId={contextId}&name=Api1&workspaceId={workspaceId}&contextType={contextType}`

Without context type, it will create for the page.
`/api/v1/import?type=CURL&contextId={contextId}&name=Api1&workspaceId={workspaceId}`

### Client changes
- Integrate api changes for curl import. Updated request params type and interfaces for the saga functions
- Updated the form value types for the curl import editor

#### PR fixes following issue(s)
Fixes [[Task]: Curl Import isn't working.](https://github.com/appsmithorg/appsmith/issues/30933)

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [x] Manual (using postman)
- [x] JUnit (existing test cases should work)

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Enhanced API import functionality with a new context-aware mechanism, allowing for more flexible integration within different parts of the application.
	- Updated various components and services to support the new contextId and contextType parameters for improved data handling and redirection logic.

### CODE REVIEW

File: app/client/src/api/ImportApi.ts [hunk 1]
```diff
@@ -16,12 +17,13 @@
...
   ): Promise<AxiosPromise<ApiResponse>> {
-    const { curl, name, pageId, workspaceId } = request;
+    const { contextId, contextType, curl, name, workspaceId } = request;
     return Api.post(CurlImportApi.curlImportURL, curl, {
       type: "CURL",
-      pageId,
+      contextId,
       name,
       workspaceId,
+      contextType,
```
[the remark this answers is not in the document]
Comment (ayushpahwa) [reply app/client/src/api/ImportApi.ts:27]: Not needed, all areas have context type as required

File: app/client/src/pages/Editor/APIEditor/helpers.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import { submitCurlImportForm } from "actions/importActions";
+import type { ActionParentEntityTypeInterface } from "@appsmith/entities/Engine/actionHelpers";
 
 export interface curlImportFormValues {
   curl: string;
-  pageId: string;
+  contextId: string;
   name: string;
+  contextType: ActionParentEntityTypeInterface;
```
[the remark this answers is not in the document]
Comment (ayushpahwa) [reply app/client/src/pages/Editor/APIEditor/helpers.ts:8]: Not needed, all areas have context type as required

**Review by nidhi-nair — APPROVED** [review]

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**